packages/fungible-token/bridge-fungible-token/tests/harness.rs

```diff
@@ -266,6 +267,141 @@ mod success {
+
+        // Relay the test message to the test contract
+        let receipts = env::relay_message_to_contract(
+            &wallet,
+            message_inputs[0].clone(),
+            contract_inputs,
+            &coin_inputs[..],
+            &env::generate_variable_output(),
+        )
+        .await;
+
+        let log_decoder = test_contract.log_decoder();
+        let refund_registered_event = log_decoder
+            .decode_logs_with_type::<RefundRegisteredEvent>(&receipts)
+            .unwrap();
+
+        // Verify the message value was received by the test contract
+        let test_contract_balance = provider
+            .get_contract_asset_balance(test_contract.contract_id(), AssetId::default())
+            .await
+            .unwrap();
+        let balance = wallet
+            .get_asset_balance(&AssetId::new(*test_contract_id.hash()))
+            .await
+            .unwrap();
+
+        assert_eq!(test_contract_balance, 100);
+        assert_eq!(
+            refund_registered_event[0].amount,
+            Bits256(env::encode_hex(config.overflow_2))
+        );
+        assert_eq!(
+            refund_registered_event[0].asset,
+            Bits256::from_hex_str(wrong_token_value).unwrap()
+        );
+        assert_eq!(
+            refund_registered_event[0].from,
+            Bits256::from_hex_str(FROM).unwrap()
+        );
+
+        // verify that no tokens were minted for message.data.to
+        assert_eq!(balance, 0);
+
+        // verify that trying to claim funds from
+        // the correct token fails
+        let error_response = test_contract
+        .methods()
+        .claim_refund(
+            Bits256::from_hex_str(FROM).unwrap(),
+            Bits256::from_hex_str(BRIDGED_TOKEN).unwrap(),
+        )
+        .call()
+        .await;
+        assert!(error_response.is_err());
+
+        let call_response = test_contract
+            .methods()
+            .claim_refund(
+                Bits256::from_hex_str(FROM).unwrap(),
+                Bits256::from_hex_str(wrong_token_value).unwrap(),
+            )
+            .call()
+            .await
+            .unwrap();
+
+        // verify correct message was sent
+        let message_receipt = call_response
+            .receipts
+            .iter()
+            .find(|&r| matches!(r, Receipt::MessageOut { .. }))
+            .unwrap();
+
+        assert_eq!(
+            *test_contract_id.hash(),
+            **message_receipt.sender().unwrap()
+        );
+        assert_eq!(
+            &Address::from_str(BRIDGED_TOKEN_GATEWAY).unwrap(),
+            message_receipt.recipient().unwrap()
+        );
+        assert_eq!(message_receipt.amount().unwrap(), 0);
+        assert_eq!(message_receipt.len().unwrap(), 104);
+
+        // message data
+        let (selector, to, token, amount) =
+            env::parse_output_message_data(message_receipt.data().unwrap());
+        assert_eq!(selector, env::decode_hex("0x53ef1461").to_vec());
```

Comment:
What hex is this? Should we factor this out into a variable / constant?

Reply:
This is the EVM function selector for `finalizeWithdrawal` on the L1 bridge contract.

```js
/// @notice Finalizes the withdrawal process from the Fuel side gateway contract
    /// @param to Account to send withdrawn tokens to
    /// @param tokenId ID of the token being withdrawn from Fuel
    /// @param amount Amount of tokens to withdraw
    /// @dev Made payable to reduce gas costs
    function finalizeWithdrawal(
        address to,
        address tokenId,
        uint256 amount
    ) external payable whenNotPaused onlyFromPortal {
        require(amount > 0, "Cannot withdraw zero");
        bytes32 fuelTokenId = messageSender();

        //reduce deposit balance and transfer tokens (math will underflow if amount is larger than allowed)
        _deposits[tokenId][fuelTokenId] = _deposits[tokenId][fuelTokenId] - amount;
        IERC20Upgradeable(tokenId).safeTransfer(to, amount);

        //emit event for successful token withdraw
        emit Withdrawal(bytes32(uint256(uint160(to))), tokenId, fuelTokenId, amount);
    }
```

In short, the L2 contract receives this message, which is taken up to the FuelMessagePortal, and executes the data payload, which is a function call to the FuelERC20Gateway contract that holds the funds in L1.